addons/sale_stock/models/sale_order.py

```diff
@@ -587,6 +585,17 @@ def _action_launch_stock_rule(self, previous_product_uom_qty=False):
                 line.name, line.order_id.name, line.order_id.company_id, values))
         if procurements:
             self.env['procurement.group'].run(procurements)
+
+        # This next block is currently needed only because the scheduler trigger is done by picking rather than stock.move
+        orders = self.mapped('order_id')
+        for order in orders:
+            pickings_to_assign = order.picking_ids.filtered(
+                lambda p: (p.state == 'confirmed' or (p.state in ['waiting', 'assigned'] and not p.printed))
+                and (p.picking_type_id.reservation_method == 'at_confirm'
+                or (p.picking_type_id.reservation_method == 'by_date' and p.scheduled_date - timedelta(days=p.picking_type_id.reservation_days_before) <= fields.Datetime.now())))
+            if pickings_to_assign:
+                # Trigger the Scheduler for Pickings
+                pickings_to_assign.action_confirm()
```

Comment:
This code is not really for assign anymore since it will be runs in _action_confirm now. However we still need to call _trigger_scheduler but it's not related to the condition upper

Reply:
Updated the name to `pickings_to_confirm` and made condition:
`pickings_to_confirm = order.picking_ids.filtered(lambda p: p.state not in ['cancel', 'done'])`